### ISSUE TITLE
Changeling Rework - Phase 1, Design

### ISSUE THREAD

**Issue by ghost** [d3athrow/vgstation13#23312]
This marks the beginning of the changeling rework to add it back into the game. First, let's talk design.

To figure out how to design a good antagonist, we need to first break down a bad antagonist. I think we can all agree Vampire is the epitome of miserable times for the victim. Flash-glare is an annoying combo that, in effect, turns the victim into a NPC. The vampire will silently go about the everlasting routine of disabling suit sensors, sucking his victim dry, then tossing a dead corpse into space. If we were to replace the victim with a NPC, there would be almost no difference. So, what do we learn from the Vampire?

- An antagonist should promote interaction with the crew.
- An antagonist should create interesting situations that can often be unique to a round.
- An antagonist should have multiple avenues of attack/ways to be an antagonist.
- **An antagonist should make the round interesting for everyone.**

So, keep these points in mind as you read the following idea.

Also note some inspiration was drawn from the following pastebin, which holds a great deal of ideas.
https://pastebin.com/U0pFnH9L

## CHANGELING 2.0
The changeling has a primary goal - consume and incorporate the consciousnesses of sapient beings. DNA is a second thought for the nu-changeling. Instead, his focus is on Genetic Points. Also known as GBP, Genetic Bioreactionary Points. 

The changeling is, in fact, a collection of nanite-like cells capable of rearrangement and duplication. The processing power required to maintain advanced form requires intense mental focus, provided by consciousnesses of those absorbed. The more people a changeling absorbs, the stronger he can get.

The changeling absorbs through a highly invasive process that requires the changeling to take a highly aggressive form. Upon transforming into the horror form the changeling will let out a rabid screech across the station and let loose an EMP in the general vicinity, disabling radios. The horror form is hulked, hyperzined, high HP, and a fucking absolute unit. In order to absorb a target the victim must be lying down - the best way to do this is through knocking them unconscious, but be wary as dead subjects do not qualify. The process takes the better of 20 seconds. 

Horror form lasts indefinitely. However, time spent in horror form will make the transformation back to a human form longer, a process which must be done while standing still. Damage in horror form cannot be healed, which means that you will want to transform back into a human quickly after devouring targets and continue with your normal business. Time spent in human form will very slowly heal the horror form's HP, shown by a bar on their screen. 

Upon obtaining a new member of the hivemind, the changeling will be granted GP to spend. The GP is spendable on a talent tree accessible through a spell. There are a variety of paths to go down that can lead to many ways to play the antag.

Members incorporated into a hivemind are loyal to their master and reside within his body. Similar to borers, they are capable of using abilities (e.g. stingers) to defend their master while in horror form, but they lack medical features during horror form. More features for hivemind to come. One idea would be that reaching enough GP would allow for the hivemind to detach from the master and function on its own as a hiveling. 

No more than two changelings can exist on station, and one is the usual number. Changelings are capable of absorbing each other if they wish and lack the the :g channel. In order to kill a changeling, you must kill its body. Killing a changeling in horror form will cause it to turn back into a human. Gibbing a changeling in human form will cause it to dissolve into blobs of flesh, one of which functions as a controlled slime. This functions similar to becoming a cultist shade, except that cutting off the head will instead trigger the slime mode. The other scattered blobs of flesh can be broken to reveal the brains of the absorbed humans. The changeling can then try to scurry away or reclaim its hivemembers in the case that it died in some retarded way.

The endgame of the changeling is likely going to involve getting advanced horror form talents and breaking down the resistance that security has set up. However, there are other paths to take. 

## Potential rounds that could happen

Impersonation is a changeling classic - kill two people in maintenance and absorb both their bodies, then pretend to be one of them. Bonus points if that person is a head. The crew will be scurrying around to try and figure out which person is the changeling, yet the only way to figure out which person is the culprit would be to cut their head off, which you can't just do on everyone. So the changeling absorbs many bodies in secret. The constant roars strike fear into the little humans. Security grabs whoever they can and forms a blockade in the security lobby, hoping that the changeling isn't one of them. But no, he's right there. He pops horror form and demolishes everyone - welcome to the hivemind. The ERT then scuttles in and lasers him to death. Shame.

Or maybe you want to wage a civil war. The changeling captures a few select crewmembers and grabs the talent to turn them into little hiveling goblin thingies that will scurry around and build a fortress. Capable of crawling through vents, they will perform sneak attacks and drag people to home base for momma to consume. It becomes a grand war of goblin shitters versus security, which ends in the expected way of security getting shitstomped by a single goblin. But wait - the robust engineer blows open a hole behind their base, exposing the gremlins to the void of space. Organization is shattered as the changeling assumes the form of a gremlin to avoid detection. The gremlins hide in vents until 300,000 degrees of air is pumped through the system, meaning that they all get burnt. The gremlins use their hivemind hud(tm) to scutter back to their overlord if they can, and the changeling attempts a last-ditch effort to try and kill whoever it can. It gets slimed and destroyed.

**Comment by boy2mantwicethefam** [comment]
GBP is a sexual activity involving torture of the male genitalia
If you want ling improved you should also allow it to impersonate implanted people

**Comment by ghost** [comment]
It also stands for good boy points.

**Comment by BurntDevil** [comment]
Look if you want a good changeling you're gonna have to move away from The Thing. It's a stale meme. It's not an interesting antag archetype. There are many other examples of changeling in other media that would translate to SS13 much better.

Primary example of this is Deep Space 9 lings. A gelatinous blob that can morph into literally anything and anyone. This fulfills the impersonation game play but also offers a much more interesting sort of antag. A creature that could be the paramedic one second, be a goliath the next, and then the humble soda can the next would ramp up paranoria to get the same feeling without the horrible and boring game play of the silent shuffling maint succer.

Let's finally move away from this idea that the Ling has to be the Thing and go towards the path of a more creative sort of changeling. Otherwise I can't do anything but oppose bringing it back.

**Comment by brndd** [comment]
Make ling into a prop hunt antag. We don't need more silent murder antags which is what any antag with jecties to kill people without being caught kind of inherently is. There's nothing wrong with silent murder antags as such, but the problem is that it's hard to make them feel really distinct from each other. A large part of this is due to the nature of the combat system we have not leaving a lot of options for stealth murder (or murder in general) aside from stun->kill (and when you get stunned do you really care if you get killed by a toolbox or by a pair of fangs?)

Getting killed and then forced to play borer that can't leave its host sounds even *more* boring than just getting killed. If a vampire glares you at least you get to saltpost in ghostchat and can then choose to become a regular borer if you really want that.

In other words, I agree with @BurntDevil here.

**Comment by ghost** [comment]
I see your concerns. The silent shuffle antag is something I wouldn't want to touch with a 6 foot pole, hence why I think having the changeling be "loud and proud" with a roar every time it absorbs is a good compromise. 

We could always add transformation abilities to the changeling, but I'm curious to hear more on why you think the archetype is a done deal. Is it just because of the absorbing people quirk, or is it more than that?

**Comment by brndd** [comment]
"Loud and proud" is a stupid meme that just substitutes one cancer (silent shuffling maint murders) for another (regular old murderboning). We already have a plenty of antags to fill both of these categories. Any new antags should be *actually* distinct in their gameplay to justify their existence, which I don't think this design is.

The gameplay here is only really different from the antag's point of view, and even then it feels derivative of vampire (kill people, get power points, buy powers).

**Comment by BurntDevil** [comment]
It's played out because the abilities are not interesting to use or be against. The absorbing mechanic is not interesting for the same reasons. The Thing works as a movie for instilling fear and suspension but it doesn't translate to ss13 in the same manner. There is no true body horror, no real spooky elements, just some guy that murder bones with lame attacks then runs away.

At least with a ds9 ling you could be creative in your methods. Want to scare the shit out of someone? Hide in the barman shaker as vodka then explode the detective when he takes the big sip. Be the chefs butcher knife until he feels safe then give him the chop. Loud traitor? Wait in the armory as an egun and glomp them when they go to arrest. Be a potted plant until your target walks by and then you're a rockernaut. Turn into a space carp and wait for someone to wire solars.

These abilities could be gained by studying the item or person in a timed scanning sort of method. It's more interesting than hey kid, get stung gg.

That sort of creative positioning is better than a reskinned vamp.

**Comment by ValkyrieSkies** [comment]
Gonna pitch an idea I talked about a while ago in regards to a Ling rework;

• Remove stings as they are and instead add an upgrade menu where genomes can be spent for body morph abilities instead, i.e: 1 genome for arm blade, 3 genomes for a rubberised flesh ability that'll block 5 taser shots, etc
• A duplication ability that costs something like 10 genomes that allows the ling to spawn another ling using the ghost of a player they've killed.
• Maybe some sort of truly nightmarish horror form for an end game that's incredibly difficult to kill and causes Centcomm to send the Nuke code to the station.
• Maybe some mutations that are controlled by ghosts independently of the ling, like the ability to detach limbs and create player controlled simple mobs from them.
• An escape ability (2 genome cost maybe) that lets the ling become a scrambling little head thing that'll die in one hit but can ventcrawl and become a proper ling again if they can succ a corpse. Can't be used when dead.

**Comment by Caboozles** [comment]
Can we move away from antags having to be strong enough to be round enders? Tators are good because unless they really work for it nobody goes “It’s tator nuke it.”

Ling should be strong enough to be a threat, but not to the point where existing in the round means you’re food.

**Comment by ghost** [comment]
After thinking about your points, I think I have to agree that reskinning murderbone abilities is a bad idea. Changeling being able to morph into anything it sees - and function as anything it sees - will probably be doable codewise, and I like the idea a lot. I will work on concepting it more.

Should we get rid of absorbing entirely, then? What about evolution/upgrades?

**Comment by brndd** [comment]
R0B0 seemed interested in working on mimic changelings on Discord, so maybe you can get something going here.

**Comment by ghost** [comment]
After some thought, the idea of having the changeling become a master of illusion grew on me. Here is some new deviations from the initial concept (names subject to change):

- Changeling has a mana pool called "nanites". It recharges over time.

- Changeling has many illusion based abilities. Some ideas are follows:

The changeling can "scan" an item by channeling near it for a while. There is a maximum of X scanned items in your memory.

The changeling can become any scanned item, structure, or filth. This process costs X nanites and the changeling will behave identically to the given thing. The changeling is capable of manipulating itself in this form, such as pulling the trigger on a gun or using a comms console after a HoP unlocks it. In addition, the changeling has a slime form that is its "true" form, which it will enter after being gibbed or upon command to scutter through vents.

Scanned items can have a replica made of them. The replica lowers your nanite maximum by X amount while it exists and functions identically to the real deal. Replicas can be altered in some ways, such as disintegrating upon touch or containing poison that looks like water instead of water. Some spells could interact with replicas - for instance, exploding them through using a spell.

The changeling is well versed in hostile entities can create illusions with a spell. For instance, the changeling would use the illusion spell and pick the following series of popups -> blood cult -> animated -> cultist blood magnetism, which he would then target to a tile. Nanites would form a fake cultist at the location and perform the animation of being teleported in, then teleported away. Alternatively, the changeling could choose bloodcult -> static -> fake rune, which he would then plant in dorms and report to security to divert their attention after chugging 5 bottles of holy water to prove his innocence.

- Upgrades using GP can improve nanite capacity, improve spells, and unlock abilities in a talent tree form. I'm unsure how the GP will be obtained, whether it be through absorbing or a new way entirely.

**Comment by brndd** [comment]
> become the Captain's Spare
> wait until the HoS picks you up
> turn into a revolver and kill them

I love it.

What about this: rather than scanning items, the changeling absorbs them. They go into a hidden inventory that can hold up to X things.  The changeling can turn into anything in this inventory, and expel stuff at will (though doing so takes a few seconds of struggling and straining). When the changeling dies, all the items pop out of its body.

**Comment by DeityLink** [comment]
So what happened about that?
